### PR TITLE
DBZ-2151 updating ANTLR parser to respect MariaDB syntax for ALTER TA…

### PR DESCRIPTION
**Pull request by jhiza**
…BLE DROP INDEX [IF EXISTS]

### CODE REVIEW

File: debezium-ddl-parser/src/test/resources/mysql/examples/ddl_alter.sql [hunk 1]
```diff
@@ -2,10 +2,12 @@
 -- Alter Table
 alter table ship_class add column ship_spec varchar(150) first, add somecol int after start_build, algorithm=instant;
 alter table t3 add column (c2 decimal(10, 2) comment 'comment`' null, c3 enum('abc', 'cba', 'aaa')), add index t3_i1 using btree (c2) comment 'some index';
+alter table t3 add column (c4 decimal(10, 2) comment 'comment`' null, add index t3_i2 using btree (c4) comment 'some index';
 alter table t2 add constraint t2_pk_constraint primary key (1c), alter column `_` set default 1;
 alter table ship_class change column somecol col_for_del tinyint first;
 alter table ship_class drop col_for_del;
 alter table t3 drop index t3_i1;
+alter table t3 drop index t3_i2 if exists;
```
Comment (jpechane) [inline debezium-ddl-parser/src/test/resources/mysql/examples/ddl_alter.sql:10]: Thanks for the PR!
Is this example correct? It seems to me that accroding to https://mariadb.com/kb/en/drop-index/  the `ON <table>` part is mandatory.

Comment (jhiza) [reply debezium-ddl-parser/src/test/resources/mysql/examples/ddl_alter.sql]: updated PR with drop index if exists _indexname_

ON is not mandatory when used from alter table.  your syntax reference is about drop index DDL.

File: debezium-ddl-parser/src/test/resources/mysql/examples/ddl_alter.sql [hunk 2]
```diff
@@ -2,10 +2,12 @@
 -- Alter Table
 alter table ship_class add column ship_spec varchar(150) first, add somecol int after start_build, algorithm=instant;
 alter table t3 add column (c2 decimal(10, 2) comment 'comment`' null, c3 enum('abc', 'cba', 'aaa')), add index t3_i1 using btree (c2) comment 'some index';
+alter table t3 add column (c4 decimal(10, 2) comment 'comment`' null, add index t3_i2 using btree (c4) comment 'some index';
```
Comment (gunnarmorling) [inline debezium-ddl-parser/src/test/resources/mysql/examples/ddl_alter.sql:5]: A closing brace is missing here after the column definition. I'll add it.

Comment (gunnarmorling) [reply debezium-ddl-parser/src/test/resources/mysql/examples/ddl_alter.sql:5]: Hum, the `ifExists` clause hasn't been added to `alterSpecification`. Seems like the core of this change is missing actually.

**Comment by gunnarmorling** [comment]
Amended as dicussed and applied. Thanks, @jhiza.

**Comment by jhiza** [comment]
Updated if relevant: https://github.com/jhiza/debezium/commits/DBZ-2151

Otherwise, if my understanding is correct, will be in 1.2?

**Comment by gunnarmorling** [comment]
> Updated if relevant: https://github.com/jhiza/debezium/commits/DBZ-2151

I already took care of it, thanks.

> Otherwise, if my understanding is correct, will be in 1.2?

Yes, this change will be part of this week's release of 1.2.0.CR1.

**Pull request closed without merging**